src/lib/Microsoft.Health.Events/EventCheckpointing/StorageCheckpointClient.cs

```diff
@@ -161,5 +168,28 @@ public async Task PublishCheckpointAsync(string partitionId)
             Checkpoint checkpoint = _checkpoints[partitionId];
             await UpdateCheckpointAsync(checkpoint);
         }
+
+        /// <summary>
+        /// Deletes the previously recorded checkpoints if the current checkpoint blob path (corresponding to the input event hub) has changed.
+        /// </summary>
+        public async Task ResetCheckpointsAsync()
+        {
+            try
+            {
+                foreach (BlobItem blob in _storageClient.GetBlobs(states: BlobStates.All, prefix: _blobCheckpointPrefix, cancellationToken: CancellationToken.None))
+                {
+                    if (!blob.Name.Contains(_blobPath, StringComparison.OrdinalIgnoreCase))
```

Comment:
>Contains [](http://example.com/codeflow?start=35&length=8)

It would be good to add logging (info) so we can see when this is or is not occuring for troubleshooting purposes.

Reply:
Added, although it will log for each blob in that path. Which I think is okay in case we ran into issue deleting a particular one, but wanted to call out the possible bloat in logging.

Reply:
Thanks I think this is fine.  Should only happen on start up so bloat should be minimal.  It might be worth while to have a trace at the very beginning, i.e. Entering Reset Checkpoints help if there ever a scenario we blobs we thought should be deleted arent (i.e. ensure we are hitting the rest checkpoint code)/